src/newc.rs

```diff
@@ -329,6 +355,11 @@ impl Builder {
         self
     }
 
+    pub fn set_mode_flag(mut self, flag: ModeFlag) -> Builder {
+        self.mode |= u32::from(flag);
```

Comment:
I think we really need to mask out the mode file type bits since nothing prevents a user from calling `set_mode_flag` more than one:

```rust
self.mode &= !0o170000;
self.mode |= u32::from(flag);
```

Additionally, we should probably change `flag` to `file_type` and the function to `set_mode_file_type` since it is specifically setting the file type bits in the mode word.